### ISSUE TITLE
Version 0.8.7 regression

### ISSUE THREAD

**Issue by decaz** [jmcarp/flask-apispec#181]
I guess cd4592892a56521c9abdae4c97d637b493bbe304 broke the following kind of functionality:
```python
    ...
    return obj, status_code, headers
```
In case of usage of returned headers exception is raised:
`werkzeug.exceptions.BadRequestKeyError: 400 Bad Request: The browser (or proxy) sent a request that this server could not understand.`

Was this done intentionally?

FYI @sloria @AdamLeyshon

**Comment by AdamLeyshon** [comment]
Not intentionally no. For now, try pinning back at 0.8.6.

I guess there was no test coverage for that so it wasn't picked up.

Unfortunately I am no longer using restful and apispec since this took a while to get merged and I moved to RestX (RestPlus fork).

Maybe if you can provide a minimal working sample, someone here can take a look, shouldn't be too difficult or they can back out my PR.

**Comment by c-kruse** [comment]
Working Example:

app.py
```
from dataclasses import dataclass

from flask import Flask
from flask_apispec import use_kwargs, marshal_with

from marshmallow import Schema
from webargs import fields

app = Flask(__name__)


class PetSchema(Schema):
    class Meta:
        fields = ('name', 'category', 'size')


@dataclass
class Pet:
    name: str
    category: str
    size: int


@app.route('/pets')
@use_kwargs({'category': fields.Str(), 'size': fields.Str()})
@marshal_with(PetSchema(many=True))
def get_pets(**kwargs):
    pets = [Pet(name=f"pet-{i}", category="dog", size=i) for i in range(3)]
    return pets, 200, {"x-msg": "hello world"}
```


First with `flask-apispec==0.8.6`:
```
# curl -sD - -o /dev/null  http://localhost:5000/pets
HTTP/1.0 200 OK
Content-Type: application/json
Content-Length: 131
x-msg: hello world
Server: Werkzeug/1.0.0 Python/3.7.6
Date: Wed, 18 Mar 2020 20:23:58 GMT
```

Now with `flask-apispec==0.8.7`
```
# curl -sD - -o /dev/null  http://localhost:5000/pets
HTTP/1.0 200 OK
Content-Type: application/json
Content-Length: 131
Server: Werkzeug/1.0.0 Python/3.7.6
Date: Wed, 18 Mar 2020 20:25:33 GMT
```

**Comment by c-kruse** [comment]
https://github.com/jmcarp/flask-apispec/pull/184

**Comment by sloria** [comment]
Fix released in 0.8.8. Thanks @c-kruse !

**Comment by decaz** [comment]
@sloria can you release it on PyPI, please?

**Comment by sloria** [comment]
Ah, woops, the previous tag build failed due to a flake8 error. Fixed; 0.8.8 will be released shortly
